### ISSUE TITLE
azurerm_mssql_database_vulnerability_assessment_rule_baseline - Root resource was present, but now absent. - Provider produced inconsistent result after apply

### ISSUE THREAD

**Issue by artyq** [hashicorp/terraform-provider-azurerm#12989]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

terraform v0.14.10
azurerm v2.72.0

### Affected Resource(s)

* `azurerm_mssql_database_vulnerability_assessment_rule_baseline`

### Terraform Configuration Files

```resource azurerm_mssql_database_vulnerability_assessment_rule_baseline app_va1258_baseline {
  server_vulnerability_assessment_id = module.mssql.sqlserver[0].assessment_id
  database_name                      = module.mssql.db[0].sql_db.name
  rule_id                            = "VA1258"
  baseline_name                      = "default"
  baseline_result {
    result = [
      var.dbowner
    ]
  }
  depends_on = [
    module.mssql,
    null_resource.configureazsql
  ]
}
```

### Debug Output

I am unable to provide a link with the debug output as this is from an internal gitlab.

```
Error: Provider produced inconsistent result after apply
When applying changes to
module.mssql_example.azurerm_mssql_database_vulnerability_assessment_rule_baseline.app_va1258_baseline,
provider "registry.terraform.io/hashicorp/azurerm" produced an unexpected new
value: Root resource was present, but now absent.
This is a bug in the provider, which should be reported in the provider's own
issue tracker.
```

### Expected Behaviour

Acknowledge that the resource is there and exists.
### Actual Behaviour

Resource is created, but Terraform can no longer see it.

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

There is several other reports with the exact same output but for different resources.
It doesn't matter if I again run plan and apply, the same error is presented.
### References

* #0000

**Comment by artyq** [comment]
Issue https://github.com/hashicorp/terraform-provider-azurerm/issues/11488

Appears to be similar to my issue.

**Comment by aristosvo** [comment]
I'll try to reproduce it and diagnose the proble, also based on duplicate issue #11488. I'll stick with this one as it is more recent.

**Comment by artyq** [comment]
Is there any progress on this?

It usually takes 3 + attempts for Terraform to notice it and not error during apply.

**Comment by catt231** [comment]
I'm also experiencing on >=2.80.0 seems to be VA2065 it has an issue with

```
resource "azurerm_mssql_server_security_alert_policy" "alert-policy" {
  resource_group_name = azurerm_mssql_server.server.resource_group_name
  server_name         = azurerm_mssql_server.server.name
  state               = var.mssql-alert-policy.enabled
  retention_days      = var.mssql-alert-policy.retain-for
  email_addresses     = var.mssql-alert-policy.notify
}

resource "azurerm_mssql_server_vulnerability_assessment" "server-vulnerability-assessment" {
  server_security_alert_policy_id = azurerm_mssql_server_security_alert_policy.alert-policy.id
  storage_container_path          = "${data.azurerm_storage_account.audit-storage-account.primary_blob_endpoint}${data.azurerm_storage_container.audit-storage-container.name}/"

  recurring_scans {
    enabled                   = true
    email_subscription_admins = false
    emails                    = var.mssql-alert-policy.notify
  }
}

resource "azurerm_mssql_database_vulnerability_assessment_rule_baseline" "firewall-rules-baseline" {
  server_vulnerability_assessment_id = azurerm_mssql_server_vulnerability_assessment.server-vulnerability-assessment.id
  database_name = "master"
  rule_id       = "VA2065"
  baseline_name = "master"

  dynamic "baseline_result" {
    for_each = local.firewall-rules

    content {
      result = [
        baseline_result.key,
        baseline_result.value["start-ip-address"],
        baseline_result.value["end-ip-address"]
      ]
    }
  }
}
```

**Comment by jamesw4** [comment]
I'm starting to wonder if the root of this really is the provider and not Azure. I've enabled trace logging and reproduced the issue. What I'm seeing is. 

- Put request to Azure API for the baseline resource (including my correct payload).
- HTTP 200 response to the put.
- A get request for the same baseline resource.
- A HTTP 404 response to the get. 

Cross checking the correlation id from the trace log to the activity log in azure I see that the action for this baseline is reported as "started" and then also a separate event for "succeeded".

**Comment by jamesw4** [comment]
Going to raise ticket with MS. In my scenario we are setting 6 baselines, randomly but quiet frequently upon create on or two would get the error mention in the first post.

As a test I added dependencies in TF just to test if them being done in sequence vs parallel had any impact. It seems in my case it does, have not been able to get it to fail. 

Then went and tested with the az PowerShell module. Same findings, doing the 6 of them in parallel resulted sometimes in one or two of them not being created despite PowerShell reporting all were created ok.

**Comment by krohm** [comment]
@jamesw4 , any update on this?

**Comment by jamesw4** [comment]
Will close as doesn’t seem specific to TF. In the end I worked around this using dependencies so they created in sequence rather than in parallel. I think MS has maybe sorted since as when I tried to reproduce the issue weeks later again with PowerShell I couldn’t reproduce the issue anymore. Thanks.

**Comment by jsmith-speedeon** [comment]
I'm hitting this in April of 2023.    It seems that applying multiple baselines to the same data warehouse sku  (in my case) causes the above error:

[Error: Provider produced inconsistent result after apply
│
│ When applying changes to azurerm_mssql_database_vulnerability_assessment_rule_baseline.spdsqlserver-va1248-baseline, provider
│ "provider[\"registry.terraform.io/hashicorp/azurerm\"]" produced an unexpected new value: Root resource was present, but now
│ absent.]

If I apply multiple times, it eventually applies all the baselines.  But there is definitely some kind of race condition it would seem.

**Comment by jamesw4** [comment]
Since changed jobs, hit by this issue again. Used same work around of chaining dependencies between each baseline resource so they are not created in parallel.

**Comment by katbyte** [comment]
Thank you for taking the time to report this issue, and apologies that it has sat without a resolution for so long.

This bug was reported against an early major version of the AzureRM provider. The provider has since moved on to v5.x, and the affected functionality has often been significantly reworked across those major releases, so bug reports from legacy major versions prior to v4.0 are generally no longer actionable as written.

As part of a cleanup of historical issues we are closing bug reports that target legacy provider versions. If you are still running into this behaviour on the current major version, please open a new issue with up-to-date reproduction details (feel free to reference this one) so it can be triaged with current context.

Thank you again for the report, and for your patience.
